spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -164,6 +177,22 @@ private boolean useHiveContext() {
     return Boolean.parseBoolean(getProperty("zeppelin.spark.useHiveContext"));
   }
 
+  public boolean diagnosis() {
+    if (getProperty("zeppelin.spark.diagnosis") == null) {
+      return false;
+    } else {
+      return Boolean.parseBoolean(getProperty("zeppelin.spark.diagnosis"));
+    }
+  }
+
+  public boolean isYarnMode() {
+    return getProperty("master").equals("yarn-client");
```

Comment:
I think "yarn-cluster" is not supported in Zeppelin. I'm not sure about the value "yarn". Could you provide some link about it?